static/js/containers/SearchPage.js

```diff
@@ -271,7 +301,8 @@ const mapDispatchToProps = (dispatch: Dispatch<*>, ownProps: Props) => ({
   getChannel: async () => {
     const channelName = getChannelName(ownProps)
     await dispatch(actions.channels.get(channelName))
-  }
+  },
+  dispatch: dispatch
```

Comment:
I think you could just have `toggleUpvote: toggleUpvote(dispatch)` here if you wanted to simplify a little bit

Reply:
hmm, weird...I tried it out and it seemed to work ok:

```diff
diff --git a/static/js/containers/SearchPage.js b/static/js/containers/SearchPage.js
index cd0b0a17..e1c2b665 100644
--- a/static/js/containers/SearchPage.js
+++ b/static/js/containers/SearchPage.js
@@ -170,7 +170,8 @@ export class SearchPage extends React.Component<Props, State> {
       initialLoad,
       total,
       dispatch,
-      upvotedPost
+      upvotedPost,
+      toggleUpvote
     } = this.props
     const { from, upvotedPosts } = this.state
 
@@ -201,7 +202,7 @@ export class SearchPage extends React.Component<Props, State> {
           <SearchResult
             key={i}
             result={result}
-            toggleUpvote={toggleUpvote(dispatch)}
+            toggleUpvote={toggleUpvote}
             upvotedPost={
               result.object_type === "post"
                 ? upvotedPosts.get(result.post_id)
@@ -302,7 +303,7 @@ const mapDispatchToProps = (dispatch: Dispatch<*>, ownProps: Props) => ({
     const channelName = getChannelName(ownProps)
     await dispatch(actions.channels.get(channelName))
   },
-  dispatch: dispatch
+  toggleUpvote: toggleUpvote(dispatch)
 })
 
 export default R.compose(
```

it's kind of a nit though, so feel free to leave as-is